Sources/FluidAudio/TextToSpeech/Kokoro/KokoroModel.swift

```diff
@@ -0,0 +1,740 @@
+    private static let baseURL = "https://huggingface.co/FluidInference/kokoro-82m-coreml/resolve/main"
+
+    /// Download file from URL if needed
+    private static func downloadFileIfNeeded(filename: String, urlPath: String) async throws {
+        let cacheDir = try TtsModels.cacheDirectoryURL()
+        let kokoroDir = cacheDir.appendingPathComponent("Models/kokoro")
+
+        // Create directory if needed
+        try FileManager.default.createDirectory(at: kokoroDir, withIntermediateDirectories: true)
+
+        let localURL = kokoroDir.appendingPathComponent(filename)
+
+        guard !FileManager.default.fileExists(atPath: localURL.path) else {
+            logger.info("File already exists: \(filename)")
+            return
+        }
+
+        logger.info("Downloading \(filename)...")
+        print("  Downloading \(filename)...")
+        let downloadURL = URL(string: "\(baseURL)/\(urlPath)")!
+
+        let (data, response) = try await URLSession.shared.data(from: downloadURL)
+
+        guard let httpResponse = response as? HTTPURLResponse,
+            httpResponse.statusCode == 200
+        else {
+            throw TTSError.modelNotFound("Failed to download \(filename)")
+        }
+
+        try data.write(to: localURL)
+        logger.info("Downloaded \(filename) (\(data.count) bytes)")
+    }
+
+    /// Download model files if needed
+    private static func downloadModelIfNeeded() async throws {
+        let cacheDir = try TtsModels.cacheDirectoryURL()
+        let modelDir = cacheDir.appendingPathComponent("Models/kokoro")
+
+        // Create Models/kokoro directory if needed
+        try FileManager.default.createDirectory(at: modelDir, withIntermediateDirectories: true)
+
+        logger.info("Model directory: \(modelDir.path)")
+        print("Model directory: \(modelDir.path)")
+
+        let modelPath = modelDir.appendingPathComponent("kokoro_completev21.mlmodelc")
+
+        if FileManager.default.fileExists(atPath: modelPath.path) {
+            logger.info("Model already downloaded")
+            print("Model already downloaded")
+            return
+        }
+
+        logger.info("Downloading kokoro_completev21 model...")
+        print("Downloading kokoro_completev21 model...")
+
+        // Create model directory
+        try FileManager.default.createDirectory(at: modelPath, withIntermediateDirectories: true)
+
+        // Download the compiled mlmodelc files (some optional depending on packaging)
+        let filesToDownload = [
+            "coremldata.bin",
+            "metadata.json",              // optional
+            "model.mil",
+            "weights/weight.bin",        // optional
+            "analytics/coremldata.bin",  // optional
+        ]
+
+        // Create subdirectories
+        try FileManager.default.createDirectory(
+            at: modelPath.appendingPathComponent("weights"),
+            withIntermediateDirectories: true
+        )
+        try FileManager.default.createDirectory(
```

Comment:
similar comments to the ones in TtsModel, these are duplicates from DownloadUtils?